### PR TITLE
Added support for `BIGINT UNSIGNED` type in MySQL and MariaDB DBTypes

### LINKED ISSUE #762: DataFrame fails on simple actions with casting BigInteger to Long
DataFrame fails on simple actions with casting BigInteger to Long.

For example, the MySQL performance schema table_handles table is defined as:

```
show create table performance_schema.table_handles;

CREATE TABLE `table_handles` (
  `OBJECT_TYPE` varchar(64) NOT NULL,
  `OBJECT_SCHEMA` varchar(64) NOT NULL,
  `OBJECT_NAME` varchar(64) NOT NULL,
  `OBJECT_INSTANCE_BEGIN` bigint unsigned NOT NULL,
  `OWNER_THREAD_ID` bigint unsigned DEFAULT NULL,
  `OWNER_EVENT_ID` bigint unsigned DEFAULT NULL,
  `INTERNAL_LOCK` varchar(64) DEFAULT NULL,
  `EXTERNAL_LOCK` varchar(64) DEFAULT NULL,
  PRIMARY KEY (`OBJECT_INSTANCE_BEGIN`),
  KEY `OBJECT_TYPE` (`OBJECT_TYPE`,`OBJECT_SCHEMA`,`OBJECT_NAME`),
  KEY `OWNER_THREAD_ID` (`OWNER_THREAD_ID`,`OWNER_EVENT_ID`)
) ENGINE=PERFORMANCE_SCHEMA DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci
```

In the dataframe:

```
DataFrame.getSchemaForSqlTable(dbConfig, "table_handles")

OBJECT_TYPE: String
OBJECT_SCHEMA: String
OBJECT_NAME: String
OBJECT_INSTANCE_BEGIN: Long
OWNER_THREAD_ID: Long?
OWNER_EVENT_ID: Long?
INTERNAL_LOCK: String?
EXTERNAL_LOCK: String?
```

Read the SQL table:

```
val dfLocksTable = DataFrame.readSqlTable(dbConfig, "table_handles")

...next cell

dfLocksTable.filter { OWNER_THREAD_ID != null }
```

results in error:

```
class java.math.BigInteger cannot be cast to class java.lang.Long (java.math.BigInteger and java.lang.Long are in module java.base of loader 'bootstrap')
java.lang.ClassCastException: class java.math.BigInteger cannot be cast to class java.lang.Long (java.math.BigInteger and java.lang.Long are in module java.base of loader 'bootstrap')
	at Line_24_jupyter._DataFrameType3_OWNER_THREAD_ID(Line_24.jupyter.kts:19)
	at Line_30_jupyter$res30$1.invoke(Line_30.jupyter.kts:6)
	at Line_30_jupyter$res30$1.invoke(Line_30.jupyter.kts:6)
	at org.jetbrains.kotlinx.dataframe.api.FilterKt.filter(filter.kt:38)
	at Line_30_jupyter.<init>(Line_30.jupyter.kts:6)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at kotlin.script.experimental.jvm.BasicJvmScriptEvaluator.evalWithConfigAndOtherScriptsResults(BasicJvmScriptEvaluator.kt:122)
	at kotlin.script.experimental.jvm.BasicJvmScriptEvaluator.invoke$suspendImpl(BasicJvmScriptEvaluator.kt:48)
	at kotlin.script.experimental.jvm.BasicJvmScriptEvaluator.invoke(BasicJvmScriptEvaluator.kt)
	at kotlin.script.experimental.jvm.BasicJvmReplEvaluator.eval(BasicJvmReplEvaluator.kt:49)
	at org.jetbrains.kotlinx.jupyter.repl.impl.InternalEvaluatorImpl$eval$resultWithDiagnostics$1.invokeSuspend(InternalEvaluatorImpl.kt:127)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:277)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:95)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:69)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:48)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at org.jetbrains.kotlinx.jupyter.repl.impl.InternalEvaluatorImpl.eval(InternalEvaluatorImpl.kt:127)
	at org.jetbrains.kotlinx.jupyter.repl.impl.CellExecutorImpl$execute$1$result$1.invoke(CellExecutorImpl.kt:79)
	at org.jetbrains.kotlinx.jupyter.repl.impl.CellExecutorImpl$execute$1$result$1.invoke(CellExecutorImpl.kt:77)
	at org.jetbrains.kotlinx.jupyter.repl.impl.ReplForJupyterImpl.withHost(ReplForJupyterImpl.kt:758)
	at org.jetbrains.kotlinx.jupyter.repl.impl.CellExecutorImpl.execute(CellExecutorImpl.kt:77)
	at org.jetbrains.kotlinx.jupyter.repl.execution.CellExecutor$DefaultImpls.execute$default(CellExecutor.kt:12)
	at org.jetbrains.kotlinx.jupyter.repl.impl.ReplForJupyterImpl.evaluateUserCode(ReplForJupyterImpl.kt:581)
	at org.jetbrains.kotlinx.jupyter.repl.impl.ReplForJupyterImpl.access$evaluateUserCode(ReplForJupyterImpl.kt:136)
	at org.jetbrains.kotlinx.jupyter.repl.impl.ReplForJupyterImpl$evalEx$1.invoke(ReplForJupyterImpl.kt:439)
	at org.jetbrains.kotlinx.jupyter.repl.impl.ReplForJupyterImpl$evalEx$1.invoke(ReplForJupyterImpl.kt:436)
	at org.jetbrains.kotlinx.jupyter.repl.impl.ReplForJupyterImpl.withEvalContext(ReplForJupyterImpl.kt:417)
	at org.jetbrains.kotlinx.jupyter.repl.impl.ReplForJupyterImpl.evalEx(ReplForJupyterImpl.kt:436)
	at org.jetbrains.kotlinx.jupyter.messaging.IdeCompatibleMessageRequestProcessor$processExecuteRequest$1$response$1$1.invoke(IdeCompatibleMessageRequestProcessor.kt:140)
	at org.jetbrains.kotlinx.jupyter.messaging.IdeCompatibleMessageRequestProcessor$processExecuteRequest$1$response$1$1.invoke(IdeCompatibleMessageRequestProcessor.kt:139)
	at org.jetbrains.kotlinx.jupyter.execution.JupyterExecutorImpl$Task.execute(JupyterExecutorImpl.kt:42)
	at org.jetbrains.kotlinx.jupyter.execution.JupyterExecutorImpl$executorThread$1.invoke(JupyterExecutorImpl.kt:82)
	at org.jetbrains.kotlinx.jupyter.execution.JupyterExecutorImpl$executorThread$1.invoke(JupyterExecutorImpl.kt:80)
	at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
```

almost any actions does this no matter if working with the numeric type or not.

### PR DESCRIPTION
**Pull request by AndreiKingsley**
Fixes #762.

### CODE REVIEW

**Comment by Jolanrensen** [comment]
could you add tests for this specific fix? :)

(the build should not have failed, I'll rerun)
(okay, maybe rebase on master, I've limited the maven logs, they were getting way too large)

**Comment by AndreiKingsley** [comment]
@Jolanrensen added tests as an initiating of #1736.

**Comment by Jolanrensen** [comment]
@AndreiKingsley that's good :) though, I would also recommend running a test with the actual database, like here https://github.com/Kotlin/dataframe/blob/c3742a95684d53a813d978a3c895090ba684c46e/dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/local/mysqlTest.kt (though, you'd need to set up a local database for that. Just ask Junie to do that for you ;P, I think @zaleslaw would appreciate it :) )

**Comment by AndreiKingsley** [comment]
@Jolanrensen , added `BIGINT UNSIGNED` columns  to Genereal MariaDB / MySQL tests!

File: dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                10,
+                javaClassName,
+                false,
+            )
+    }
+
+    class MariaDBTypes {
+
+        object BIGINT_UNSIGNED : ColumnType(
+            "BIGINT UNSIGNED",
```
Comment (Jolanrensen) [inline dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt:37]: I'd recommend using named arguments. Makes refactoring easier :)

File: dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+@RunWith(Enclosed::class)
+class JdbcTypesTest {
+
+    abstract class ColumnType(
+        val sqlTypeName: String,
+        val jdbcType: Int,
+        val javaClassName: String,
+        val expectedKotlinType: KType,
+    ) {
+        fun mockkColMetaData() =
```
Comment (Jolanrensen) [inline dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt:23]: since when does "mock" have 2 k's? ;P

File: dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            TableColumnMetadata(
+                "name",
+                sqlTypeName,
+                jdbcType,
+                10,
+                javaClassName,
+                false,
```
Comment (Jolanrensen) [inline dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt:30]: hmm by setting this to false always, you might not catch the cases where someone forgets to take care about nullability (which is an easy mistake to make). Maybe we should test both isNullable and !isNullable. This could also be done in a next pr of course

File: dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                MariaDb.getExpectedJdbcType(type.mockkColMetaData()) shouldBe type.expectedKotlinType
+            }
+        }
+    }
+
+    class MySqlDBTypes {
+
+        object BIGINT_UNSIGNED : ColumnType(
+            "BIGINT UNSIGNED",
+            20,
```
Comment (Jolanrensen) [inline dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt:59]: I'd use Sql.Types for this

Comment (Jolanrensen) [reply dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt:59]: oh, it doesn't exist?... interesting

File: dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt (hunk 2 above)
Comment (Jolanrensen) [reply dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt:23]: oh and "metadata" is one word, so it would become: "mockColMetadata()"

File: dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt (hunk 4 above)
Comment (AndreiKingsley) [reply dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt:59]: Yep, it's a MySQL/MariaDB specific type.

File: dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt (hunk 2 above)
Comment (AndreiKingsley) [reply dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt:23]: Because I often use [MockK](https://github.com/mockk/mockk) variables usually called `mockk..` 😆

File: dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt (hunk 4 above)
Comment (Jolanrensen) [reply dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/db/jdbcTypesTest.kt:59]: classic

**Comment by zaleslaw** [comment]
Good fix. It solves a real bug and the added coverage goes in the right direction.

A few action items:

- Rename mockkColMetaData() to something like createColumnMetadata(). It is a test helper, not really a mock.
- Use named arguments when creating `TableColumnMetadata(...)` to make tests easier to read.
- Add nullable and non-nullable cases in `JdbcTypesTest`.
- Add a couple of boundary values for `BIGINT UNSIGNED`, not only type mapping checks.
- Consider a small shared helper to reduce duplication between MySQL and MariaDB tests.

Also, this PR seems bigger than a one-case fix. It highlights a more general gap in JDBC type mapping tests. This looks like the start of a broader testing strategy for JDBC types, and that can help us find more mismatches and missing cases in the future. But in Beta-6.

**Comment by AndreiKingsley** [comment]
@zaleslaw will do all of this as a part of #1736.
